components/pedigree/resources/src/main/resources/pedigree/controller.js

```diff
@@ -261,6 +261,55 @@ define([
                 }
             }
 
+            // Ask if the patient should be a part of a study
+            var templateSelectionContainer = new Element('div');
+            var templateSelectionForm = new Element('div', {'class' : 'xform'});
+            var formDataWrapper = new Element('div');
+            templateSelectionForm.insert(formDataWrapper);
+
+            formDataWrapper.insert(templateSelectionContainer);
+            var submitButton = new Element('input', {type : 'button', value : "Submit", 'class' : 'button'});
+            formDataWrapper.insert(submitButton);
+            submitButton.observe('click', function(event) {
+                selectedStudies = $$('.study-selection-dialog  input[type="radio"]');
+                var studyName = "";
+                selectedStudies.each( function(item) {
+                    if (item.checked) {
+                        studyName = item.value;
+                    }
+                });
+                callPatientCreate(studyName);
+                modalBox.closeDialog();
+            });
+
+            var cancelButton = new Element('input', {type : 'button', value : "Cancel", 'class' : 'button secondary'});
+            formDataWrapper.insert(' ').insert(cancelButton);
+            var modalBox = new PhenoTips.widgets.ModalPopup(templateSelectionForm, false, {'title': "Study selection", 'removeOnClose': false, 'extraClassName': "study-selection-dialog"});
+            cancelButton.observe('click', function(event) {
+                modalBox.closeDialog();
+            });
+
+            var getStudyDialogURL = editor.getExternalEndpoint().getStudyBindingService();
+            new Ajax.Request(getStudyDialogURL, {
+                onCreate : function() {
+                  //templateSelectionContainer.update('<img src="$xwiki.getSkinFile('icons/xwiki/ajax-loader-large.gif')"/>');
+                  modalBox.showDialog();
+                },
+                onSuccess : function(response) {
+                   templateSelectionContainer.update(response.responseText);
```

Comment:
this is "OK" for now, but using HTML from PhenoTips is a not a very good idea, since
 a) we have to reply on a certain HTML structure to parse the list of available studies
 b) (less important) this adds another dependence on specifics of XWiki code to pedigree

I think both problems can be avoided when unrelated necessary fix descrobed below is implemented

Reply:
unrelated problem is that this dialog pops up even if there are no studies configured in the PT instance, which is the case for most standalone out-of-the box instances. So regardless of everything else a check if there are studies should be done in `https://github.com/phenotips/phenotips/blob/master/components/pedigree/ui/src/main/resources/PhenoTips/PedigreeInterface.xml` and passed to the pedigree editor, and study selection dialog should only be shown if ther eis at least one study. But if that has to be done, might as well pass the list of all study names o the pedigree, in which case we do not need a `StudyBindingService` any more, and a simple dialogue with a simple list of radio buttons or a simple select can be used instead